src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -0,0 +1,258 @@
+    periscopeStorageInfo: PeriscopeStorage,
+    aksPeriscopeFilePath: string
+) {
+
+    // persicope file is written now, replace the acc name and keys.
+    const replace = require("replace");
+
+    const base64Name = Buffer.from(periscopeStorageInfo.storageName).toString('base64');
+    const base64Sas = Buffer.from(periscopeStorageInfo.storageDeploymentSas).toString('base64');
+    replace({
+        regex: "# <accountName, base64 encoded>",
+        replacement: `"${base64Name}"`,
+        paths: [aksPeriscopeFilePath],
+        recursive: true,
+        silent: true,
+    });
+
+    replace({
+        regex: "# <saskey, base64 encoded>",
+        replacement: `"${base64Sas}"`,
+        paths: [aksPeriscopeFilePath],
+        recursive: true,
+        silent: true,
+    });
+}
+
+export async function generateDownloadableLinks(
+    periscopeStorage: PeriscopeStorage,
+    output: string
+): Promise<PeriscopeHTMLInterface[] | undefined> {
+
+    try {
+        const storageAccount = periscopeStorage.storageName;
+        const storageKey = periscopeStorage.storageKey;
+        const sas = periscopeStorage.storageDeploymentSas;
+        const sevenDaySas = periscopeStorage.sevenDaysSasyKey;
+
+        // Get DNS Core hostname which Periscope use it as name of the container.
+        const matches = output.match(/(https?:\/\/[^\s]+)/g);
+        let containerName;
+        if (matches![0].indexOf('://') !== -1) {
+            containerName = matches![0].replace('https://', '').split('.')[0];
+        }
+
+        // Use SharedKeyCredential with storage account and account key
+        const sharedKeyCredential = new StorageSharedKeyCredential(storageAccount, storageKey);
+        const blobServiceClient = new BlobServiceClient(
+            `https://${storageAccount}.blob.core.windows.net`,
+            sharedKeyCredential
+        );
+
+        // Hide this all under single pupose funct which lik elike : getZipDir or get me dirname
+        const containerClient = blobServiceClient.getContainerClient(containerName);
+
+        // List all current blob.
+        const listCurrentUploadedFolders = [];
+        for await (const item of containerClient.listBlobsByHierarchy("/")) {
+            if (item.kind === "prefix") {
+                listCurrentUploadedFolders.push(item.name);
+            }
+        }
+
+        // Sort and get the latest uploaded folder under the container used by periscope.
+        const periscopeHtmlinterfaceList = [];
+        for await (const blob of containerClient.listBlobsFlat()) {
+            // Get the latest uploaded folder, then Identify the Zip files in the latest uploaded logs within that folder
+            // and extract *.zip files which are individual node logs.
+            const periscopeHTMLInterface = <PeriscopeHTMLInterface>{};
+            if (blob.name.indexOf(listCurrentUploadedFolders.sort().reverse()[0]) !== -1 && blob.name.indexOf('.zip') !== -1) {
```

Comment:
It's not obvious what the intent is here: could we pull this out into something with an intention-revealing name?  (Whether a function or a local variable.)